src/simulation/LoadGenerator.cpp

```diff
@@ -668,6 +787,29 @@ LoadGenerator::getNextAvailableAccount()
     return sourceAccountId;
 }
 
+// Returns account ID for the next available account that has not already set up
+// a contract instance
+uint64_t
+LoadGenerator::getNextAvailableAccountForSorobanSetup()
```

Comment:
what is different about contract instances that we can't use `getNextAvailableAccount` here? I might not have enough soroban background, but why do we have 1-1 mapping of accounts vs contract instances?

Reply:
A 1-1 mapping has a few advantages. The way we charge resource fees, every TX pays for every resource it uses. This means that if 2 TXs access the same entry, they both get charged for the read even though core only does one load. By having each account have it's own instance, you guarantee that each TX will have no shared state such that we have no cached data. Additionally, each contract instance has it's own key space, so having unique contract instance greatly simplifies IO loads since we can use the same SCVal keys for every TX but they still refer to separate `LedgerKeys` such that they require separate loads. I could just have a pool of available contract instances instead of having a one-to-one mapping, but the setup overhead is not very significant (it just takes one TX per instance), and with a one-to-one mapping we can reuse the existing `getNextAvailableAccount` functionality without having to reimplement something like `getNextAvailableInstance`.

Reply:
During setup we want each account to submit a single deploy instance TX. If we use `getNextAvailableAccount`, we will get an available account that may or may not have already deployed an instance. `getNextAvailableAccountForSorobanSetup` is just an extension of `getNextAvailableAccount` that makes sure the returned account does not yet have a contract instance deployed.